daffodil-test/src/test/resources/org/apache/daffodil/section23/dfdl_expressions/expressions.tdml

```diff
@@ -7184,6 +7185,11 @@
     <tdml:infoset><tdml:dfdlInfoset><div21>7.290005321703885E-5</div21></tdml:dfdlInfoset></tdml:infoset>
   </tdml:parserTestCase>
 
+  <tdml:parserTestCase name="div22" root="div22" model="XPathDiv" description="Section 23 - DFDL Expressions - div"
+    config="cfg_errorOnResultCoercion">
+    <tdml:document></tdml:document>
+    <tdml:infoset><tdml:dfdlInfoset><div22>0.028125</div22></tdml:dfdlInfoset></tdml:infoset>
```

Comment:
I'm not sure if this test fully tests that long div long returns a decimal. If long div long return a double, we will allow automatic upcasting to the expected xs:decimal type, so this doesn't really show that we're doing the right thing, I think. This test will probably work without the above change.

Maybe the best way to handle this is to add a unit test somewhere (maybe TestDFDLExpressionEvaluation.scala) that evaluates an expression and checks the return type is the correct thing.

Reply:
New updates look good, but this could use some tests to make sure we're doing the right thing. This is a corner case that really needs tests to ensure we're doing the right thing.

Reply:
Just pushed up the tests. Meant to do that last week, but forgot about adding tests then. I was struggling with getting the unit tests working to verify the types, so I did what you originally suggested and had tests expecting incorrect types and then made sure the appropriate error message was displayed.